Mono.Debugging/Mono.Debugging.Client/StackFrame.cs

```diff
@@ -122,63 +123,98 @@ internal void Attach (DebuggerSession debugSession)
 		/// <summary>
 		/// Gets the full name of the stackframe. Which respects Session.EvaluationOptions.StackFrameFormat
 		/// </summary>
-		public virtual string FullStackframeText {
-			get {
-				var methodNameBuilder = new StringBuilder (this.SourceLocation.MethodName);
-				var options = session.EvaluationOptions.Clone ();
-				if (options.StackFrameFormat.ParameterValues) {
-					options.AllowMethodEvaluation = true;
-					options.AllowToStringCalls = true;
-					options.AllowTargetInvoke = true;
-				} else {
-					options.AllowMethodEvaluation = false;
-					options.AllowToStringCalls = false;
-					options.AllowTargetInvoke = false;
-				}
+		[Obsolete]
+		public string FullStackframeText {
+			get { return GetFullStackFrameText (); }
+		}
 
-				var args = this.GetParameters (options);
+		public string GetFullStackFrameText ()
+		{
+			return GetFullStackFrameText (session.EvaluationOptions);
+		}
 
-				//MethodName starting with "["... it's something like [ExternalCode]
-				if (!this.SourceLocation.MethodName.StartsWith ("[", StringComparison.Ordinal)) {
-					if (options.StackFrameFormat.Module && !string.IsNullOrEmpty (this.FullModuleName)) {
-						methodNameBuilder.Insert (0, System.IO.Path.GetFileName (this.FullModuleName) + "!");
-					}
-					if (options.StackFrameFormat.ParameterTypes || options.StackFrameFormat.ParameterNames || options.StackFrameFormat.ParameterValues) {
-						methodNameBuilder.Append ("(");
-						for (int n = 0; n < args.Length; n++) {
-							if (args [n].IsEvaluating) {
-								var manualReset = new ManualResetEvent (false);
-								EventHandler updated = (s, e) => {
-									manualReset.Set ();
-								};
-								args [n].ValueChanged += updated;
-								if (args [n].IsEvaluating)
-									if (!manualReset.WaitOne (2000))
-										session.OnDebuggerOutput (true, "Timeout evaluating parameters for StackFrame.");
-								args [n].ValueChanged -= updated;
-							}
-							if (n > 0)
-								methodNameBuilder.Append (", ");
-							if (options.StackFrameFormat.ParameterTypes) {
-								methodNameBuilder.Append (args [n].TypeName);
-								if (options.StackFrameFormat.ParameterNames)
-									methodNameBuilder.Append (" ");
+		public virtual string GetFullStackFrameText (EvaluationOptions options)
```

Comment:
Just pass in a cancellationtoken in EvaluationOptions and use that?

Reply:
This would avoid hardcoding a 2s timer here, which could probably not be enough. Also, shouldn't we allow only the async versino to be override-able?

Reply:
2 good points.

Reply:
FWIW, the old code used a 2s timer which is where I got that from.

Reply:
Should we also drop the possibility to override the non-async version? Otherwise we could end up with 2 different implementations based on the API you call.

Reply:
Well, the .NET Core backend overrides the non-async version, so it's still needed.